provider/keycloak_openid_client.go

```diff
@@ -211,7 +298,17 @@ func resourceKeycloakOpenidClientUpdate(data *schema.ResourceData, meta interfac
 		return err
 	}
 
-	setOpenidClientData(data, client)
+	var serviceAccountUserId string
+
+	if client.ServiceAccountsEnabled {
+		serviceAccountUser, err := keycloakClient.GetOpenidClientServiceAccountUserId(client.RealmId, client.Id)
+		if err != nil {
+			return err
+		}
+		serviceAccountUserId = serviceAccountUser.Id
+	}
+
+	setOpenidClientData(data, client, serviceAccountUserId)
```

Comment:
This block of code (lines 301 - 311) seems to be repeated multiple times.  Could you update the `setOpenidClientData` to do this instead?

Reply:
Done